### ISSUE TITLE
using UTF-8 encoding on raw files(even when specified that they should do that).

### ISSUE THREAD

**Issue by decembre** [MrAlex94/Waterfox-Old#63]
### What happened?

Waterfox Classic 2020.11 / 56.6

I tested on a new profile of Waterfox Classic, the install of an userstyle With Stylish (the last version without privacy pb) and without Stylus addon instaled.
by example for this userstyle:
<a href="https://userstyles.world/style/248/flickr-widescreen-bigone-v-187">Flickr WideScreen - BigONE</a>
Because Stylish don't handle the install and open its own addon page in Chrome Store (?), 
When i  click on install button, it open a page 
https://userstyles.world/api/style/248.user.css

I copy paste the CSS in my Old Stylish editor, cut the to header:
/* ==UserStyle==
... / ...
==/UserStyle== */

The userstyles work, but strangely, it convert some Unicode, like that:

From:
.context-slider-scrappy-view .context-slider > .nav-l span:before {
    content: " ◄" ;
}

To this:

.context-slider-scrappy-view .context-slider > .nav-l span:before {
    content: " â—„" ;
}

Can you enable UTF-8 encoding on raw files?

It's a duplicate of a report i make here:
<a href="https://github.com/userstyles-world/userstyles.world/issues/107">Unicode caracters wrong formated when i test install without Stylus WaterfoxCo/Waterfox#107 [Github Stylus]</a> where dev said it's related to a bug in Waterfox.

### Reproducible?

- [ ] I have tested this on Firefox.

### Version

G3.𝑋 (Current)

### What platform are you seeing the problem on?

Windows

**Comment by reallyuniquename** [comment]
TL;DR: Not a Waterfox issue. Use View - Encoding - Unicode as a workaround.

There is no byte order mark in that CSS file, there is no `@charset` attribute in the sheet and that server doesn't specify an encoding in its HTTP headers. Waterfox falls back to your system's default encoding which is ANSI. It even explicitly tells you so `The character encoding of the plain text document was not declared. The document will render with garbled text in some browser configurations if the document contains characters from outside the US-ASCII range`.

**Comment by decembre** [comment]
<code>Not a Waterfox issue</code>
Thea's right:
After a test with Stylus deactivated,
Quantum do the same thing...

Thanks for the workaround:
<blockquote>Use View - Encoding - Unicode as a workaround.</blockquote>

I trying that before posting here, but i can't find a way to do it:
<s>Where i can change this setting?</s>
Edit after a re test more careful, i found how (I trying before with the contextual menu or in the Firefox options.):

**Top** menu > Affichage > Encodage du Texte > Unicode

Choose "Auto" is not enough.
But it seems permanent .
If permanent, no problem after for other type of views?

In an other side, it's difficult to know when it is useful to change this type.
By example:
<code>.context-slider-scrappy-view .context-slider > .nav-l span:before {
content: " ◄" ;</code>

is located line ~ 15000... and it was only because i know the result, i can suppose a problem.
Without that, i can, by mistake, copy paste the code CSS to publish it.

<blockquote>no @charset</blockquote>
But, in the stylus dev's reply,said:
<blockquote>(even when specified that they should do that)</blockquote>

The raw file as an ".user.css" in its url.
So that's not enough?
It's not possible to  detect it and serve it with UTF-8 encoding ?
Re edit:
If Stylus disable, the view type is marked "Occidental"
**After Stylus enable, without reload the page, it is marked as  "Unicode"**
Can use this type of detection for your own project?

in <a href="https://support.mozilla.org/fr/kb/changer-ce-que-fait-firefox-lorsque-vous-cliquez-telechargez-fichier">Changer ce que Firefox fait lorsque vous cliquez sur un fichier ou le téléchargez</a>
Mozilla recommend for these types of problems:
<a href="https://addons.mozilla.org/fr/firefox/addon/content-type-fixer/">Content-Type Fixer</a>
Tested:
Can be installed but Not working effectively with Waterfox <57....
:-(

Anyway, thanks for the work on Waterfox Classic:
It is so practical to have something working at the OLD way!
And for sure, give the time to find how to pass the "superior" level 
without lost too many things and discover newer.
;-)

**Comment by reallyuniquename** [comment]
> In an other side, it's difficult to know when it is useful to change this type.

You can always use File - Save As, open it in notepad and then copy paste it into old Stylish editor. That should work most of the time.

> After Stylus enable, without reload the page, it is marked as "Unicode"

Maybe Stylus itself handles `*.user.css` and forces encoding to UTF-8. 

>It's not possible to detect it and serve it with UTF-8 encoding ?

It's possible to serve it with UTF-8 encoding header but that should be done by whoever owns userstyles.world, e.g. https://www.w3.org/International/questions/qa-css-charset#http

**Comment by decembre** [comment]
Thanks for your explanation.
I  had sent this request to userstyles.world before...
<a href="https://github.com/userstyles-world/userstyles.world/issues/107">Unicode caracters wrong formated when i test install without Stylus #107 [userstyles.world - GitHub]</a>
;-)
They suggest to address it to .... Waterfox.
Finally, maybe with your idea can be done by them.

**Comment by decembre** [comment]
Good news:
Fixed in userstyle.world:
<a href="https://github.com/userstyles-world/userstyles.world/issues/107">[now no error UTF-8 encoding in the user.css.](https://github.com/userstyles-world/userstyles.world/issues/107)</a>
